docs/DeveloperGuide.adoc

```diff
@@ -364,6 +387,144 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+        if backup storage file is present:
+            retrieve back up addressbook
+            save back up addressbook content to addressbook
+        else:
+            retrieve sample addressbook
+    } catch (addressbook has incorrect format) {
+        create empty addressbook
+    } catch (unreadable addressbook) {
+        create empty addressbook
+    }
+}
+----
+
+Back up file is kept up-to-date by updating itself when an AddressBookChangedEvent is raised.
+
+=== Favorite
+
+The Favorite function is created for frequent users who have have certain "close contacts" and
+want the contacts on top of their list for the ease of access.
+In order to implement such feature, two things need to be implemented:
+
+* Favorite class +
+* Sorting
+
+==== Favorite class
+A favorite class is implemented and used by Person class. A person object does not construct with a favorite object but it is
+able to setFavorite a new favorite object.
+A favorite object is different from other attribute object is that it just contains a simple boolean property.
+
+To favorite a person in the list, a favorite command is implemented. The favorite command follows the same parser logic
+as `delete` command, which is to parse the index of the person to favorite. Hence, the favoriteCommand also extends
+UndoableCommand and overrides the methods in UndoableCommand.
+
+Model and UI components also have to be updated. For model, a method toggle favorite is implemented to favorite or "defavorite"
+a person at specified index:
+
+[source,java]
+----
+public void favorite(ReadOnlyPerson toFavorite)
+    throws PersonNotFoundException {
+        requireNonNull(toFavorite);
+        int index = internalList.indexOf(toFavorite);
+
+        if (index == -1) {
+            throw new PersonNotFoundException();
+        }
+
+        Person newPerson = new Person(toFavorite);
+        newPerson.setFavorite(new Favorite(!toFavorite.getFavorite().favorite));
+
+        internalList.set(index, newPerson);
+        sort();
+}
+----
+
+And for UI, a simple label is created and `setVisible` to `true` for any index that is
+favorited. The labels are `setGraphic` to followed image:
+
+image::heart.png[width="40"]
+
+==== Sorting
+
+A sorting algorithm needs to be implemented for the `UniquePersonList` of the `Model` Component.
+Fortunately, it can be easily done with `List.sort()` and  `Java predicate`. The priority of sorting is such that
+the persons favorited always appear on the top of the list, then all persons w/out favorited are
+sorted by alphabetical order:
+
+[source,java]
+----
+private void sort() {
+        internalList.sort((ReadOnlyPerson p1, ReadOnlyPerson p2) -> {
+            if (!p1.getFavorite().equals(p2.getFavorite())) {
+                return p2.getFavorite().getValue() - p1.getFavorite().getValue();
+            } else {
+                return p1.getName().fullName.compareTo(p2.getName().fullName);
+            }
+        });
+}
+----
```

Comment:
As a general rule of thumb, try not to put code here. Use sequence diagrams or other forms of modelling to explain your implementation. See the undo/redo implementation description for reference.